src/endless/Eosldr.cpp

```diff
@@ -0,0 +1,204 @@
+#include "stdafx.h"
+#include "Eosldr.h"
+#include "GeneralCode.h"
+#include "WMI.h"
+
+// Paths for g2ldr binaries, used to chainload GRUB from the Windows Boot
+// Manager on BIOS systems, rather than the other way round.
+#define EOSLDR_SUBDIRECTORY             L"eosldr"
+// Stub loaded by WBM
+#define EOSLDR_MBR_FILE                 L"eosldr.mbr"
+// GRUB core.img with a special preamble, loaded by eosldr.mbr
+#define EOSLDR_IMG_FILE                 L"eosldr"
+
+// GUID for Endless OS's entry in the BCD boot menu, stored in the Uninstall
+// registry key
+#define REGKEY_BCD_GUID         L"EndlessBcdGuid"
+
+bool EosldrInstaller::CanInstall(const CString & bootFilesPath)
+{
+    FUNCTION_ENTER_FMT("%ls", bootFilesPath);
+
+    return !!PathFileExists(bootFilesPath + EOSLDR_SUBDIRECTORY);
+}
+
+static bool CopyAndHide(const CString &source, const CString &target)
```

Comment:
Good point, I didn't really think about that… it would be nice to avoid the annoying ACL and the argument you give is a valid one for doing so :wink: